computestorage/format.go

```diff
@@ -18,9 +46,31 @@ func FormatWritableLayerVhd(ctx context.Context, vhdHandle windows.Handle) (err
 	defer span.End()
 	defer func() { oc.SetSpanStatus(span, err) }()
 
-	err = hcsFormatWritableLayerVhd(vhdHandle)
+	h := vhdHandle
+	// On RS5 HcsFormatWritableLayerVhd expects to receive a disk handle instead of a vhd handle.
+	if osversion.Build() < osversion.V19H1 {
+		if err := vhd.AttachVirtualDisk(syscall.Handle(vhdHandle), vhd.AttachVirtualDiskFlagNone, &vhd.AttachVirtualDiskParameters{Version: 1}); err != nil {
+			return err
+		}
+		defer func() {
+			if detachErr := vhd.DetachVirtualDisk(syscall.Handle(vhdHandle)); err == nil && detachErr != nil {
+				err = detachErr
+			}
+		}()
+		diskPath, err := vhd.GetVirtualDiskPhysicalPath(syscall.Handle(vhdHandle))
+		if err != nil {
+			return err
+		}
+		diskHandle, err := openDisk(diskPath)
+		if err != nil {
+			return err
+		}
+		defer windows.CloseHandle(diskHandle) // nolint: errcheck
```

Comment:
why do we need to close this handle but we don't close the handle that's created if this section of code doesn't run?

Reply:
We don't handle closing the vhd handle passed in, it's expected the caller will close it. This function owns this handle essentially so we need to close it ourselves if we hit this path.

Reply:
Ah okay, makes sense